manifests/ca.pp

```diff
@@ -89,27 +89,27 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
-define openvpn::ca(
-  $country,
-  $province,
-  $city,
-  $organization,
-  $email,
-  $common_name  = 'server',
-  $group        = false,
-  $ssl_key_size = 2048,
-  $ca_expire    = 3650,
-  $key_expire   = 3650,
-  $key_cn       = '',
-  $key_name     = '',
-  $key_ou       = '',
-  $tls_auth     = false,
+define openvpn::ca (
+  String $country,
+  String $province,
+  String $city,
+  String $organization,
+  String $email,
+  String $common_name     = 'server',
+  Optional[String] $group = undef,
+  Integer $ssl_key_size   = 2048,
+  Integer $ca_expire      = 3650,
+  Integer $key_expire     = 3650,
+  String $key_cn          = '',
+  String $key_name        = '',
+  String $key_ou          = '',
+  Boolean $tls_auth       = false,
 ) {
 
   include openvpn
 
   $group_to_set = $group ? {
```

Comment:
Is this actually needed? Can't we just assign $openvpn::params::group in the params section above_ (line 99)

Reply:
sadly yes (or at least i did not manage to do it) because if i do it like this:
"Optional[String] $group                   = $::openvpn::params::group,"
ill get an error like:
"Evaluation Error: Unknown variable: '::openvpn::params::group'"

Reply:
if you do have a better idea, i am glad to change it!

Reply:
ahh, because this is a defined type. yeah let's stick with this logic.